xtask/src/swift.rs

```diff
@@ -85,7 +100,92 @@ fn build_library() -> Result<()> {
     Ok(())
 }
 
-fn build_xcframework() -> Result<()> {
-    println!("XCFramework not yet implemented.");
+fn build_for_target(target: &str, release: bool) -> Result<PathBuf> {
+    let (release_flag, rel_type_dir) =
+        if release { ("--release", "release") } else { ("", "debug") };
+    cmd!("cargo build -p matrix-sdk-ffi {release_flag} --target {target}").run()?;
+    Ok(workspace::root_path()?
+        .join("target")
```

Comment:
You can choose to have a different target directory through cargo configuration. The more robust way of getting the target directory is fetching `target_directory` from `cargo metadata` output the same way we are getting the `workspace_root` in the `root_path` fn right now.